services/helloworld/src/Site.hs

```diff
@@ -114,18 +120,37 @@ tokenOk :: Maybe ByteString -> Handler App App Bool
 tokenOk Nothing = return False
 tokenOk (Just token) = do
     hwConfig <- gets aHWConfig
-    let sid = serviceId hwConfig
-        key = serviceKey hwConfig
-        url = thentosBackendUrl hwConfig <> "/session/" <> urlEncode token <> "/login/" <> urlEncode sid
+    let sid = encodeUtf8 $ serviceId hwConfig
```

Comment:
I'd rather not. If I did that, than I'd have to `decodeUtf8` it in a bunch of places. At least `encodeUtf8` is total.